lib/mindwendel/lanes.ex

```diff
@@ -78,17 +79,33 @@ defmodule Mindwendel.Lanes do
           asc: lane.inserted_at
         ]
 
-    Repo.all(lane_query)
+    ideas_query = Idea
+    ideas_advanced_query = build_ideas_query_with_filter(ideas_query, filters[:idea_labels])
+
+    lane_query
+    |> Repo.all()
     |> Repo.preload(
-      ideas: [
-        :link,
-        :likes,
-        :label,
-        :idea_labels
-      ]
+      ideas:
+        ideas_advanced_query
+        |> preload([
+          :link,
+          :likes,
+          :idea_labels
```

Comment:
Ah no, not like this! I also don't think this works. The correct way is:

```
Repo.preload(ideas: {ideas_query, [:link, :likes, :idea_labels]})
```

https://hexdocs.pm/ecto/Ecto.Repo.html#c:preload/3-examples (last example)

Reply:
ok, I will change it. but just for info: the other works as well, also saw it in some blog post

Reply:
```
lane_query
    |> Repo.all()
    |> Repo.preload(
      ideas: {ideas_advanced_query, [
        :link,
        :likes,
        :idea_labels
      ]}
    )
    ```

gives me:

> invalid preload `{Mindwendel.Brainstormings.Idea, [:link, :likes, :idea_labels]}` in `[ideas: {Mindwendel.Brainstormings.Idea, [:link, :likes, :idea_labels]}]`. preload expects an atom, a (nested) keyword or a (nested) list of atoms